spec/unit/dotnet_agent_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'fauxhai'
+
+describe 'appdynamics::dotnet_agent' do
+  context 'win2008r2' do
+    before do
+      Fauxhai.mock('platform' => 'windows', 'version' => '2008R2')
+    end
+    let(:chef_run) do
+      ChefSpec::SoloRunner.new do |node|
+        node.automatic['kernel']['os_info']['windows_directory'] = 'c:\windows'
+        node.automatic['kernel']['machine'] = 'x86_64'
+        node.set['appdynamics']['dotnet_agent']['template']['cookbook'] = 'appdynamics'
+        node.set['appdynamics']['dotnet_agent']['template']['source'] = 'dotnet/setup.config.erb'
+        node.set['appdynamics']['app_name'] = 'myapp'
+        node.set['appdynamics']['controller']['host'] = 'appdynamics-controller.domain.com'
+        node.set['appdynamics']['controller']['port'] = '443'
+        node.set['appdynamics']['controller']['user'] = 'someuser'
+        node.set['appdynamics']['controller']['accesskey'] = 'supersecret'
+        node.set['appdynamics']['http_proxy']['host'] = 'appdynamics-proxy.domain.com'
+        node.set['appdynamics']['http_proxy']['port'] = '80'
+        node.set['appdynamics']['install_dir'] = 'C:\Program Files\Appdynamics'
+      end.converge(described_recipe)
+    end
+    it 'creates a temp directory' do
+      expect(chef_run).to create_directory('c:\\windows\\Temp')
+      expect(chef_run).to_not create_directory('c:\\windows\\not_temp')
```

Comment:
Not sure I see the point of these negative expectations. They will always succeed unless ChefSpec/RSpec are completely broken (or if we strangely make a directory called precisely `not_temp`). If ChefSpec/RSpec are completely broken, there's no reason to trust that the `to_not` expectation is correct or otherwise.

Am I thinking about this the wrong way? Right now, I think these negative assertions don't provide any additional coverage or guarantee of the cookbook's functionality and should be removed.